synapse/lib/stormlib/infosec.py

```diff
@@ -0,0 +1,267 @@
+  'S': {'U': 6.42, 'C': 7.52},
+  'C': {'N': 0, 'L': 0.22, 'H': 0.56},
+  'I': {'N': 0, 'L': 0.22, 'H': 0.56},
+  'A': {'N': 0, 'L': 0.22, 'H': 0.56},
+  'E': {'X': 1, 'U': 0.91, 'P': 0.94, 'F': 0.97, 'H': 1},
+  'RL': {'X': 1, 'O': 0.95, 'T': 0.96, 'W': 0.97, 'U': 1},
+  'RC': {'X': 1, 'U': 0.92, 'R': 0.96, 'C': 1},
+  'CR': {'X': 1, 'L': 0.5, 'M': 1, 'H': 1.5},
+  'IR': {'X': 1, 'L': 0.5, 'M': 1, 'H': 1.5},
+  'AR': {'X': 1, 'L': 0.5, 'M': 1, 'H': 1.5},
+}
+
+vect2prop = {
+    'AV': 'cvss:av',
+    'AC': 'cvss:ac',
+    'PR': 'cvss:pr',
+    'UI': 'cvss:ui',
+    'S': 'cvss:s',
+    'C': 'cvss:c',
+    'I': 'cvss:i',
+    'A': 'cvss:a',
+    'E': 'cvss:e',
+    'RL': 'cvss:rl',
+    'RC': 'cvss:rc',
+    'CR': 'cvss:cr',
+    'IR': 'cvss:ir',
+    'AR': 'cvss:ar',
+    'MS': 'cvss:ms',
+    'MC': 'cvss:mc',
+    'MI': 'cvss:mi',
+    'MA': 'cvss:ma',
+    'MAV': 'cvss:mav',
+    'MAC': 'cvss:mac',
+    'MPR': 'cvss:mpr',
+    'MUI': 'cvss:mui',
+}
+
+def roundup(x):
+    return (math.ceil(x * 10) / 10.0)
+
+@s_stormtypes.registry.registerLib
+class CvssLib(s_stormtypes.Lib):
+    '''
+    A Storm library which implements CVSS score calculations.
+    '''
+    _storm_locals = (
+        {'name': 'calculate', 'desc': 'Calculate the CVSS score values for an input risk:vuln node.',
+         'type': {'type': 'function', '_funcname': 'calculate',
+                  'args': (
+                      {'name': 'node', 'type': 'storm:node',
+                       'desc': 'A risk:vuln node from the Storm runtime.'},
+                      {'name': 'save', 'type': 'boolean', 'default': True,
+                       'desc': 'If true, save the computed scores to the node properties.'},
+                      {'name': 'vers', 'type': 'str', 'default': '3.1',
+                       'desc': 'The version of CVSS calculations to execute.'},
+                  ),
+                  'returns': {'type': 'dict', 'desc': 'The name of the newly created backup.', }
+        }},
+        {'name': 'vectToProps', 'desc': 'Parse a CVSS vector and return a dictionary of risk:vuln props.',
+         'type': {'type': 'function', '_funcname': 'vectToProps',
+                  'args': (
+                      {'name': 'text', 'type': 'str', 'desc': 'A CVSS vector string.'},
+                  ),
+                  'returns': {'type': 'dict', 'desc': 'A dictionary of risk:vuln secondary props.', }
+        }},
+        {'name': 'saveVectToNode', 'desc': 'Parse a CVSS vector and record properties on a risk:vuln node.',
+         'type': {'type': 'function', '_funcname': 'saveVectToNode',
+                  'args': (
+                      {'name': 'node', 'type': 'storm:node',
+                       'desc': 'A risk:vuln node to record the CVSS properties on.'},
+                      {'name': 'text', 'type': 'str', 'desc': 'A CVSS vector string.'},
+                  ),
+                  'returns': {'type': 'dict', 'desc': 'A dictionary of risk:vuln secondary props.', }
```

Comment:
Doesn't look like this one actually has a return value.

Reply:
I'll check this